core/thread.c

```diff
@@ -55,6 +55,47 @@ const char *thread_getname(kernel_pid_t pid)
 #endif
 }
 
+void thread_zombify(void)
+{
+    if (irq_is_in()) {
+        return;
+    }
+
+    unsigned state = irq_disable();
+    sched_set_status((thread_t *)sched_active_thread, STATUS_ZOMBIE);
+    irq_restore(state);
+    thread_yield_higher();
+
+    sched_task_exit();
```

Comment:
This line should never be reached, right? The stack might not be valid anymore.

Should we maybe replace it with sth like `__builtin_unreachable()`?

At least there needs to be a comment.

Reply:
or `UNRACHABLE()`?

Reply:
yes, line 69: `sched_task_exit();` should never be reached.

I think it is a good idea to change it to `UNREACHABLE()` /  `__builtin_unreachable()` . To do that we have to change the function to a non-void function. (at least on my maschine)

Reply:
> To do that we have to change the function to a non-void function. (at least on my maschine)

What compiler / architecture is that?